lib/travis/guest-api/app/endpoint/steps.rb

```diff
@@ -8,44 +8,58 @@ class Steps < Travis::GuestApi::App::Endpoint
     end
 
     post '/steps' do
-      steps = env['rack.parser.result']
-      steps = [ params ] unless (Array === env['rack.parser.result'])
+      begin
+        steps = env['rack.parser.result']
+        steps = [ params ] unless (Array === env['rack.parser.result'])
+
+        steps.map! do |step|
+
+          halt 422, {
```

Comment:
I would agree with `halt 422` rather than work by exception. In addition, we could tighten the `begin-rescue` so that it only encloses the code which throws `AddStepException`.